crates/dojo-types/src/primitive.rs

```diff
@@ -70,7 +70,15 @@ impl Primitive {
             Primitive::U128(_)
             | Primitive::ContractAddress(_)
             | Primitive::ClassHash(_)
-            | Primitive::Felt252(_) => Ok(format!("'0x{:064x}'", value[0])),
+            | Primitive::Felt252(_) => {
+                let hex_string = format!("{:x}", value[0]);
+                let trimmed_hex = hex_string.trim_start_matches('0');
+                Ok(if trimmed_hex.is_empty() {
+                    "'0x0'".to_string()
+                } else {
+                    format!("'0x{}'", trimmed_hex)
+                })
```

Comment:
I believe we want to continue storing the prefixed hex values (for lexical sorting) but want to remove the prefix in the graphql api before returning it to the user

Reply:
Right, should happen on graphql side

@tarrencev can we finish reviewing this https://github.com/dojoengine/dojo/pull/1005 and merge so this fix can be built on top?

Reply:
Ah right, my baaaad.
Should be it now, let me know if I can do better!

Reply:
Works great!

![image](https://github.com/dojoengine/dojo/assets/610224/e204802d-c587-4eca-b150-01b7e02b67b3)